src/test/java/org/broadinstitute/hellbender/tools/walkers/haplotypecaller/HaplotypeCallerIntegrationTest.java

```diff
@@ -815,24 +818,58 @@ public void testMaxAlternateAlleles(final String bam, final String reference, fi
             if ( getNumAltAllelesExcludingNonRef(vc) > maxAlternateAlleles ) {
-                sawTooManyAllelesWithoutMaxAlternateArg = true;
+                final List<Allele> mostLikelyAlleles = AlleleSubsettingUtils.calculateMostLikelyAlleles(vc, HomoSapiensConstants.DEFAULT_PLOIDY, maxAlternateAlleles);
+                expectedSubsettedAllelesByLocus.put(new SimpleInterval(vc), mostLikelyAlleles);
             }
         }
-        Assert.assertTrue(sawTooManyAllelesWithoutMaxAlternateArg,
+
+        // Then assert that we saw at least one call with more than maxAlternateAlleles alt alleles
+        // when running without --max-alternate-alleles (otherwise, the tests below won't be meaningful):
+        Assert.assertTrue(! expectedSubsettedAllelesByLocus.isEmpty(),
                 "Without --max-alternate-alleles, there should be at least one call in the output with more than " + maxAlternateAlleles +
                         " alt alleles in order for this test to be meaningful");
 
-        // Then, assert that in the VCF produced WITH --max-alternate-alleles, there are no calls with
-        // more than maxAlternateAlleles alt alleles, excluding NON_REF
+        // Now assert that in the VCF produced WITH --max-alternate-alleles, there are no calls with
+        // more than maxAlternateAlleles alt alleles, excluding NON_REF. Also check each call that would
+        // have had more than maxAlternateAlleles alleles against the expected list of subsetted alleles,
+        // to ensure that we selected the most likely alleles:
         for ( final VariantContext vc : callsWithMaxAlternateAlleles ) {
+
+            // No call should have more than the configured number of alt alleles (excluding NON_REF)
             Assert.assertTrue(getNumAltAllelesExcludingNonRef(vc) <= maxAlternateAlleles,
                     "Number of alt alleles exceeds --max-alternate-alleles " + maxAlternateAlleles + " for VariantContext: " + vc);
+
+            // If there's an entry for this locus in our table of expected alleles post-subsetting, assert
+            // that we selected the right alleles during subsetting.
+            List<Allele> alleleSubsettingExpectedResult = expectedSubsettedAllelesByLocus.get(new SimpleInterval(vc));
+            if ( alleleSubsettingExpectedResult != null ) {
+
+                // CollectionUtils.isEqualCollection() will compare the lists of Alleles without
+                // regard to ordering
+                Assert.assertTrue(CollectionUtils.isEqualCollection(vc.getAlleles(), alleleSubsettingExpectedResult),
+                        "For call " + vc + " expected alleles after subsetting were: " + alleleSubsettingExpectedResult +
+                                 " but instead found alleles: " + vc.getAlleles());
+            }
+
+            // For completeness sake, also check the genotypes to ensure that no genotypes reference
+            // an allele not present in the VC:
```

Comment:
I'm pretty sure that htsjdk will throw an exception if there is an allele in a genotype that's not in the VC.  Don't ask me know I know.  (ahem)  So this is probably redundant.

Reply:
Well, that's most likely true in practice, since `VariantContextBuilder` always requests the `VariantContext.Validation.GENOTYPES` validation when you set the `genotypes()`. But there is a way to bypass that check by calling `genotypesNoValidation()` on the builder instead of `genotypes()`, so I figure we might as well keep this sanity check around in case someone decides to do something crazy with a `VariantContextBuilder` one day...